reframe/core/pipeline.py

```diff
@@ -569,24 +596,24 @@ def pipeline_hooks(cls):
     #:     </sanity_functions_reference>`) as values.
     #:     :class:`None` is also allowed.
     #: :default: :class:`None`
-    perf_patterns = fields.TypedField(
-        typ.Dict[str, _DeferredExpression], type(None))
+    perf_patterns = _var('/', typ.Dict[str, _DeferredExpression],
+                         type(None), value=None)
 
     #: List of modules to be loaded before running this test.
     #:
     #: These modules will be loaded during the :func:`setup` phase.
     #:
     #: :type: :class:`List[str]`
     #: :default: ``[]``
-    modules = fields.TypedField(typ.List[str])
+    modules = _var('/', typ.List[str], value=[])
 
     #: Environment variables to be set before running this test.
     #:
     #: These variables will be set during the :func:`setup` phase.
     #:
     #: :type: :class:`Dict[str, str]`
     #: :default: ``{}``
-    variables = fields.TypedField(typ.Dict[str, str])
+    variables = _var('/', typ.Dict[str, str], value={})
```

Comment:
We need to deprecate this and change it to `env_variables` or something like that.

Reply:
That's for 4.0 😄 It will be sooo disruptive. Nonetheless, I suggest opening an issue about it. We can collect such disruptive changes to bundle them for the next major bump.

Reply:
Issue created: https://github.com/eth-cscs/reframe/issues/1774